### PR TITLE
Add block editor support

### LINKED ISSUE #10: Check for Gutenberg compatibility
WordPress 5.0 will be here soon. Time for us to check, if our plugins are compatible with the Gutenberg Editor and WP 5.0 (a beta version is yet to be published). 

- [x] check (Gutenberg) editor with this plugin
- [ ] check plugin settings site with active Gutenberg
- [ ] fix potential issues
- [ ] add `Tested up to: 5.0` to the plugins readme
- [ ] publish changes as minor update or with upcoming plugin update

### PR DESCRIPTION
**Pull request by florianbrinkmann**

### CODE REVIEW

**Comment by florianbrinkmann** [comment]
Working on the PHPCS issues…

**Comment by florianbrinkmann** [comment]
First, I had one notice per error, but then I noticed that in the Classic Editor case there is only one notice with all errors, so I updated it accordingly to only use one notice in Gutenberg.

File: assets/js/notice.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+( function() {
+	wp.domReady( function() {
```
Comment (patrickrobrecht) [inline assets/js/notice.js:2]: Thanks @florianbrinkmann for the PR. I've just checked out the code and tried to test it. In my local development environment I just get a `TypeError: wp.domReady is not a function` error notice on the console.

Is this a problem of my setup (WP 5.4 with SPCL as only active plugin) or are any depencencies missing (such as a WP Editor JavaScript not loaded yet etc.)?

Comment (florianbrinkmann) [reply assets/js/notice.js:2]: @patrickrobrecht oh, yes, there was a dependency missing. Just tried to reproduce the error and disabled all plugins, but did not get it. Something on my local install seems to load that already 🤔 Added the dependency that should be correct and changed the `wp-core-data` dependency to `wp-data`, I guess/hope that is the right one.

Comment (patrickrobrecht) [reply assets/js/notice.js:2]: Thanks for your fast reply and the fix. In my local WordPress 5.4 it's still not working due to the variable `postId` being `null`, although executing `wp.data.select('core/editor').getCurrentPostId();` in the console leads to the correct result. Maybe there's a problem with the order of scripts are loaded?

Comment (florianbrinkmann) [reply assets/js/notice.js:2]: Hm, does that happen in all cases (adding a new post and editing an existing one)?

Comment (patrickrobrecht) [reply assets/js/notice.js:2]: Yes, the error occurred in both cases. Moving the line `postId = ...` into the function of the subscribe block fixes the issue for me (I pushed a bugfix commit to your branch). Can you verify that?

**Review by patrickrobrecht — APPROVED** [review]

File: assets/js/notice.js (hunk 1 above)
Comment (florianbrinkmann) [reply assets/js/notice.js:2]: Yes, works for me, too. Should we maybe move the declaration of the `postId` variable into the `if ( isSaving && ! isChecking ) {` to not fire it that often? I am not sure if it has a negative impact on the editor performance to get it with every change, was just an idea.

**Pull request merged**